fund.go

```diff
@@ -0,0 +1,222 @@
+func getAuthenticationKey() (string, error) {
+	if authenticationKey.Data != "" {
+		return authenticationKey.Data, nil
+	}
+	tempPath, _ := url.Parse("authentication?username=mifos&password=password")
+	tempPath1 := client.HostName.ResolveReference(tempPath).String()
+	path := strings.Replace(tempPath1, "/savingsaccounts", "", -1)
+	req, err := http.NewRequest("POST", path, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+
+	var resp *http.Response
+	errTry := highbrow.Try(5, func() error {
+		resp, err = client.HttpClient.Do(req)
+		return err
+	})
+	if errTry != nil {
+		log.Println(errTry.Error())
+		rawMessage := json.RawMessage([]byte(errTry.Error()))
+		return "", &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		rawMessage := json.RawMessage(body)
+		return "", &FineractError{GetFineractStatusCode(resp.StatusCode), &rawMessage}
+	}
+
+	err = json.Unmarshal(body, &authenticationKey)
+	if err != nil {
+		rawMessage := json.RawMessage([]byte(err.Error()))
+		return "", &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+
+	return authenticationKey.Data, err
+
+}
+
+func makeRequest(reqType, url string, payload interface{}) ([]byte, error) {
+	b, err := json.Marshal(payload)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	req, err := http.NewRequest(reqType, url, bytes.NewBuffer(b))
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+	authKey, err := getAuthenticationKey()
+	if err != nil {
+		rawMessage := json.RawMessage([]byte(err.Error()))
+		return nil, &FineractError{ErrAuthenticationFailure, &rawMessage}
+	}
+	req.Header.Set("Authorization", "Basic "+authKey)
+
+	var resp *http.Response
+	errTry := highbrow.Try(5, func() error {
+		resp, err = client.HttpClient.Do(req)
+		return err
+	})
+	if errTry != nil {
+		rawMessage := json.RawMessage([]byte(errTry.Error()))
+		return nil, &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		rawMessage := json.RawMessage(body)
+		return nil, &FineractError{GetFineractStatusCode(resp.StatusCode), &rawMessage}
+	}
+
+	return body, err
+}
+
+func (client *Client) FundIncrement(fundId string, request FundIncrementRequest) (*FundIncrementResponse, error) {
+	tempPath, _ := url.Parse(path.Join(fundId, "transactions?command=deposit"))
+	path := client.HostName.ResolveReference(tempPath).String()
+	body, err := makeRequest("POST", path, request)
+	if err != nil {
+		return nil, err
+	}
+	var response FundIncrementResponse
+	err = json.Unmarshal(body, &response)
```

Comment:
This code is common for all methods.

Reply:
Pass response as an argument to makeRequest which accepts an interface{} to unmarshal the response into.